src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,94 @@
+package storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import exception.JarvisException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+
+public class Storage {
+    private final String dataFilePath;
+
+    /**
+     * Constructs a Storage object.
+     *
+     * @param dataFilePath Path of the file to store and retrieve data from.
+     */
+    public Storage(String dataFilePath) {
+        this.dataFilePath = dataFilePath;
+    }
+
+    /**
+     * Loads data from the file path.
+     *
+     * @return List of tasks stored in the file.
+     * @throws JarvisException If file retrieval failed.
+     */
+    public List<Task> loadData() throws JarvisException {
```

Comment:
Good implementation of "Make the happy path prominent", you put the expected behaviour first, and subsequently the potential problem path of directory not existing next.